up-l3/usubscription/v3/README.adoc

```diff
@@ -18,139 +18,615 @@ SPDX-FileType: DOCUMENTATION
+--
+
+[.specitem,oft-sid="req~usubscription-subscribe~1",oft-needs="impl,utest"]
+--
+When a client calls the `Subscribe()` function, uSubscription service *MUST* store the client as a subscriber of the provided topic for the duration of the uSubscription service runtime (no persistency across service restarts).
+--
+
+[.specitem,oft-sid="dsn~usubscription-subscribe-multiple~1",oft-needs="impl,utest"]
+--
+When a client calls the `Subscribe()` function for a topic that it is already subscribed on, uSubscription service *MUST* return a response message containing the current subscription status for this client-topic combination (e.g. `SUBSCRIBED` or `SUBSCRIBE_PENDING`).
+--
+
+[.specitem,oft-sid="req~usubscription-subscribe-notifications~1",oft-needs="impl,utest"]
+--
+When a client calls the `Subscribe()` function, uSubscription service *MUST* generate subscription change notifications that indicate any change of subscription state of the subscribed topic.
+--
+
+[.specitem,oft-sid="dsn~usubscription-subscribe-invalid-topic~1",oft-needs="impl,utest"]
+--
+When receiving a `Subscribe()` request that contains a topic that
+* is not a valid xref:../../../basics/uri.adoc[uProtocol URI] or
+* contains a _wildcard_ authority or
+* contains a _wildcard_ uEntity ID (`ue_id`) or
+* contains a _wildcard_ resource ID,
+
+a uSubscription service *MUST* return a failure status message with link:../../../up-core-api/uprotocol/v1/ucode.proto[`UCode`] `INVALID_ARGUMENT`.
+--
+
+[#unsubscribe-operation]
+=== Unsubscribe()
+
+[.specitem,oft-sid="dsn~usubscription-unsubscribe-protobuf~1",oft-needs="impl"]
+--
+The uSubscription service `Unsubscribe()` function *MUST* implement the corresponding protobuf definition in link:../up-core-api/uprotocol/core/uSubscription/v3/usubscription.proto[usubscription.proto].
+--
+
+[.specitem,oft-sid="req~usubscription-unsubscribe~1",oft-needs="impl,utest"]
+--
+When a client calls the `Unsubscribe()` function, uSubscription service *MUST* stop tracking the client as a subscriber to the topic provided with the function call.
+--
+
+[.specitem,oft-sid="dsn~usubscription-unsubscribe-multiple~1",oft-needs="impl,utest"]
+--
+When a client calls the `Unsubscribe()` function for a topic that it has not subscribed to, uSubscription service *MUST* return a response message containing the subscription status `UNSUBSCRIBED`.
+--
+
+[.specitem,oft-sid="req~usubscription-unsubscribe-notifications~1",oft-needs="impl,utest"]
+--
+When the last client subscribed to a topic calls the `Unsubscribe()` function on that topic, uSubscription service *MUST* stop generating subscription change notifications for that topic.
+--
+
+[.specitem,oft-sid="dsn~usubscription-unsubscribe-invalid-topic~1",oft-needs="impl,utest"]
+--
+When receiving a `Unsubscribe()` request that contains a topic that
+* is not a valid xref:../../../basics/uri.adoc[uProtocol URI] or
```

Comment:
The bullets are not correct, my suggestion should fix it, same for other parts of the spec